server/index.ts

```diff
@@ -1,29 +1,77 @@
-import * as http from 'http';
-import socketIo from 'socket.io';
-import * as socketFunctions from './socketFunctions';
-import { mongoose } from '@typegoose/typegoose';
-import { exit } from 'process';
-import { seedCards } from './seedCards';
-import { Shape, FillStyle, Color } from './db/card';
+import * as http from "http";
+import socketIo from "socket.io";
+import { mongoose } from "@typegoose/typegoose";
+import { exit } from "process";
+import { seedCards } from "./seedCards";
+import { Shape, FillStyle, Color } from "./db/card";
+import { RoomController } from "./controllers/roomController";
 
 const PORT = process.env.PORT || 3000;
-const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/passt';
+const MONGODB_URI =
+  process.env.MONGODB_URI || "mongodb://localhost:27017/passt";
 if (!MONGODB_URI) {
-    console.error('Please set the MONGODB_URI environment variable');
-    exit(1);
+  console.error("Please set the MONGODB_URI environment variable");
+  exit(1);
 }
 
 const app = http.createServer();
 const io = socketIo(app);
+const roomController = new RoomController();
 
-io.on('connection', (socket) => {
-    socket.on('join', (data: socketFunctions.JoinRoomData) => socketFunctions.joinRoom)
-})
+io.on("connection", (socket) => {
+  socket.on("create", (data: { open: boolean }) =>
+    roomController
+      .createRoom(data.open)
+      .then((room) => socket.emit("createdRoom", room))
+  );
+  socket.on("join", (data: { roomCode: string; playerName: string }) =>
+    roomController
+      .joinRoom(data.roomCode, socket.id, data.playerName)
```

Comment:
What happens if the room cannot be joined?

Reply:
Great question. I can emit an error, if that works